packages/utils/upgrade/resources/codemods/5.0.0/dependency-upgrade-react-and-react-dom.json.ts

```diff
@@ -0,0 +1,67 @@
+import path from 'node:path';
+import semver from 'semver';
+
+import type { modules } from '../../../dist';
+
+const REACT_DEP_NAME = 'react';
+const REACT_DEP_PATH = `dependencies.${REACT_DEP_NAME}`;
+
+const REACT_DOM_DEP_NAME = 'react-dom';
+const REACT_DOM_DEP_PATH = `dependencies.${REACT_DOM_DEP_NAME}`;
+
+const DEP_NEW_VERSION_RANGE = '^18.0.0';
+
+/**
+ * Specifically targets the root package.json and updates the react and react-dom dependency version.
+ *
+ * We first check if the react and react-dom dependency is listed in the package.json. If the dependency is
+ * found, we verify its version.
+ *
+ * If the detected version does not satisfy the new version range, we replace it with the new one.
+ *
+ * Conversely, if no react or react-dom dependency is listed, we add it with the new version range.
+ */
+const transform: modules.runner.json.JSONTransform = (file, params) => {
+  const { cwd, json } = params;
+
+  const rootPackageJsonPath = path.join(cwd, 'package.json');
+
+  if (file.path !== rootPackageJsonPath) {
+    return file.json;
+  }
+
+  const j = json(file.json);
+
+  if (j.has(REACT_DEP_PATH) && j.has(REACT_DOM_DEP_PATH)) {
```

Comment:
What if it only has one of them :thinking:

Reply:
doesn't matter since they have to be both present, if they have only one in the strapi project something is wrong, so  we set both to `18.0.0` tried to make this logic simple no need to overthink it, both need to be present and on the same version